### DISCUSSION TITLE
Rust performance build configuration

### DISCUSSION THREAD

**Discussion by jtracey** [shadow/shadow#2938] [Ideas]
Has anyone checked the [performance impact of Rust build configs](https://nnethercote.github.io/perf-book/build-configuration.html)? E.g., when running large experiments, do we know if it's worth reducing the number of compilation units, or enabling native CPU instructions? Or is it best to stick with the default build/install configs?

**Comment by stevenengler** [comment]
I tried a few configurations in #2779 with a 5% Tor network, but didn't see any obvious wins so I didn't look into it further. I think it's worth a closer look, but I don't expect there to be large gains, or at least until we have more code moved to rust.
